### PR TITLE
Shrike & Hivemind Heals no longer cancels on target movement

### PR DESCRIPTION
**Pull request by Runian**
## About The Pull Request
Shrike's Psychic Cure and Hivemind's Heal no longer cancels if the target moves; ability won't be used/go through if checks such as distance or line of sight fail.

Gives Hivemind's Heal some flavortext similar to Psychic Cure with slight differences.
Adds some missing conditionals regarding silent messages in Psychic Cure.

## Why It's Good For The Game
I feel that a healing ability that you can use at range (with the power of your mind) shouldn't be constrained into the same limitation of other healing powers where your target basically has to be doing nothing. If the target is standing still and doing nothing, then you don't really need the range given to these abilities as you would just walk up next to them and heal. In addition, makes it more useful since it opens the opportunity for healing moving crit xenos.

## Changelog
:cl:
balance: Shrike and Hivemind's healing ability no longer cancels if target moves. 
/:cl:

### CODE REVIEW

**Comment by gpinrgrmoma** [comment]
THE DAYS OF PASSIVE AGGRESSIVE POINTING AT THE QUEEN AT 5% HP WHO KEEPS JITTERSTEPPING AROUND ONLY TO DIE ARE OVER!!!!

**Comment by Breadeagle** [comment]
What about queen? Since queen heal and hivemind heal are pretty much the same (possibly literally the same if im remembering right)

**Comment by Vincent983** [comment]
its quite literally just drone acid salve, so drone, queen, and hivemind

**Comment by Runian** [comment]
> What about queen? Since queen heal and hivemind heal are pretty much the same (possibly literally the same if im remembering right)

I've changed the flavortext of Hivemind's heal to be like Shrike's flavortext. The only reason the Queen doesn't get this is because they have "acid" while Hivemind doesn't have any.

File: code/modules/mob/living/carbon/xenomorph/castes/shrike/abilities_shrike.dm [hunk 1]
```diff
@@ -232,6 +232,7 @@
 		KEYBINDING_NORMAL = COMSIG_XENOABILITY_PSYCHIC_CURE,
 	)
 	var/heal_range = SHRIKE_HEAL_RANGE
+	var/do_after_flags = IGNORE_TARGET_LOC_CHANGE
```
Comment (Lumipharon) [inline code/modules/mob/living/carbon/xenomorph/castes/shrike/abilities_shrike.dm:235]: why is this even a var? All the do_after's are separate so can't you just specify them there?

**Review by Lumipharon — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/castes/hivemind/abilities_hivemind.dm [hunk 2]
```diff
@@ -68,6 +68,29 @@
...
+	playsound(target, SFX_ALIEN_DROOL, 25)
+	new /obj/effect/temp_visual/telekinesis(get_turf(target))
+	var/mob/living/carbon/xenomorph/patient = target
+	patient.salve_healing()
+	owner.changeNext_move(CLICK_CD_RANGE)
+	succeed_activate()
+	add_cooldown()
+	if(owner.client)
+		var/datum/personal_statistics/personal_statistics = GLOB.personal_statistics_list[owner.ckey]
+		personal_statistics.heals++
```
Comment (Lumipharon) [inline code/modules/mob/living/carbon/xenomorph/castes/hivemind/abilities_hivemind.dm:92]: wuh? Why are you copypasting queen heal code?

Comment (Runian) [reply code/modules/mob/living/carbon/xenomorph/castes/hivemind/abilities_hivemind.dm:92]: I wanted to change the flavortext while keeping everything else the same. If I do ..(), then A) queen's flavortext will appear ontop of hivemind's flavor and B) there will be 2 do_afters. The only alternative I can think of (if you prefer it) is making a new proc called something like "heal_flavortext()" so I don't have to add use_ability for hivemind.

Comment (TiviPlus) [reply code/modules/mob/living/carbon/xenomorph/castes/hivemind/abilities_hivemind.dm:92]: Or just make a var you should never be copy pasting massive blocks like this to change one string

File: code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm [hunk 3]
```diff
@@ -354,13 +354,25 @@
...
-	target.visible_message(span_xenowarning("\the [owner] vomits acid over [target], mending their wounds!"))
+	if(!hivemind_heal)
+		if(!do_after(owner, 1 SECONDS, NONE, target, BUSY_ICON_FRIENDLY, BUSY_ICON_MEDICAL))
+			return FALSE
+		target.visible_message(span_xenowarning("\the [owner] vomits acid over [target], mending their wounds!"))
+	else
+		if(!do_after(owner, 1 SECONDS, IGNORE_TARGET_LOC_CHANGE, target, BUSY_ICON_FRIENDLY, BUSY_ICON_MEDICAL))
+			return FALSE
+		if(!can_use_ability(target, TRUE))
+			return FALSE
```
Comment (Lumipharon) [inline code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm:370]: you are still copy pasting stuff twice.
Just have 1 do_after, they're both the same except the loc change flag, so just make that a ternary based on your new var.
the can_use_ability proc should be checked regardless as well.

Then just make the message be behind the an if.

File: code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm [hunk 4]
```diff
@@ -354,13 +354,25 @@
 	)
 	heal_range = HIVELORD_HEAL_RANGE
 	target_flags = ABILITY_MOB_TARGET
+	var/hivemind_heal = FALSE
```
Comment (Lumipharon) [inline code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm:357]: autodoc

**Review by Lumipharon — CHANGES_REQUESTED** [review]

**Review by TiviPlus — APPROVED** [review]
@Kuro020

**Pull request merged**